packages/trpc/server/routers/viewer/eventTypes.tsx

```diff
@@ -603,4 +611,93 @@ export const eventTypesRouter = router({
         id,
       };
     }),
+  duplicate: eventOwnerProcedure.input(EventTypeDuplicateInput.strict()).mutation(async ({ ctx, input }) => {
+    const { id: originalEventTypeId, title: newEventTitle, slug: newSlug } = input;
+    const eventType = await ctx.prisma.eventType.findUnique({
+      where: {
+        id: originalEventTypeId,
+      },
+      include: {
+        customInputs: true,
+        schedule: true,
+        users: true,
+        team: true,
+        workflows: true,
+        webhooks: true,
+      },
+    });
+
+    if (!eventType) {
+      throw new TRPCError({ code: "NOT_FOUND" });
+    }
+
+    // Validate user is owner of event type or in the team
+    if (eventType.userId !== ctx.user.id) {
+      if (eventType.teamId) {
+        const isMember = await ctx.prisma.membership.findFirst({
+          where: {
+            userId: ctx.user.id,
+            teamId: eventType.teamId,
+          },
+        });
+        if (!isMember) {
+          throw new TRPCError({ code: "FORBIDDEN" });
+        }
+      }
+      throw new TRPCError({ code: "FORBIDDEN" });
+    }
+
+    const {
+      customInputs,
+      users,
+      locations,
+      team,
+      recurringEvent,
+      bookingLimits,
+      metadata,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      workflows: _workflows,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      webhooks: _webhooks,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      schedule: _schedule,
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - not typed correctly as its set on SSR
+      descriptionAsSafeHTML: _descriptionAsSafeHTML,
+      ...rest
+    } = eventType;
+
+    const data: Prisma.EventTypeCreateInput = {
+      ...rest,
+      title: newEventTitle,
+      slug: newSlug,
+      locations: locations ?? undefined,
+      team: team ? { connect: { id: team.id } } : undefined,
+      users: users ? { connect: users.map((user) => ({ id: user.id })) } : undefined,
+      recurringEvent: recurringEvent || undefined,
+      bookingLimits: bookingLimits ?? undefined,
+      metadata: metadata === null ? Prisma.DbNull : metadata,
+    };
+
+    const newEventType = await ctx.prisma.eventType.create({ data });
+
+    // Create custom inputs
+    if (customInputs) {
+      const customInputsData = customInputs.map((customInput) => {
+        const { id: _, options, ...rest } = customInput;
+        return {
+          options: options ?? undefined,
+          ...rest,
+          eventTypeId: newEventType.id,
+        };
+      });
+      await ctx.prisma.eventTypeCustomInput.createMany({
+        data: customInputsData,
+      });
+    }
+
```

Comment:
@CarinaWolli Which can be the best way to clone/copy webhooks/workflows?

Reply:
For workflows, this worked for me: 
```
    const ids = _workflows.map(workflow => workflow.workflowId);
    const connectWorkflowIds = ids.map(id => { return {workflow: { connect: { id: id } } }});
    
        const data: Prisma.EventTypeCreateInput = {
      ...rest,
      title: newEventTitle,
      slug: newSlug,
      locations: locations ?? undefined,
      team: team ? { connect: { id: team.id } } : undefined,
      users: users ? { connect: users.map((user) => ({ id: user.id })) } : undefined,
      recurringEvent: recurringEvent || undefined,
      bookingLimits: bookingLimits ?? undefined,
      metadata: metadata === null ? Prisma.DbNull : metadata,
      workflows: {
        create: connectWorkflowIds,
      }
    };
```